### PR TITLE
Add unit tests for ShardingSphereTable

### LINKED ISSUE #25361: Add unit tests for ShardingSphereTable
This issue is to add unit tests for the ShardingSphereTable class, it includes the following subtasks:

- [ ] ShardingSphereTable#containsColumn
- [ ] ShardingSphereTable#containsIndex
- [ ] ShardingSphereTable#getColumn
- [ ] ShardingSphereTable#getColumns
- [ ] ShardingSphereTable#getConstraints
- [ ] ShardingSphereTable#getIndex
- [ ] ShardingSphereTable#getIndexes
- [ ] ShardingSphereTable#putColumn
- [ ] ShardingSphereTable#putIndex
- [ ] ShardingSphereTable#removeIndex

### PR DESCRIPTION
**Pull request by 100paperkite**
Fixes #25361

Changes proposed in this pull request:
  -

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        shardingSphereTable = new ShardingSphereTable();
+    }
+    
+    @Test
+    void putColumn() {
+        ShardingSphereColumn column1 = new ShardingSphereColumn("foo_column_1", Types.INTEGER, true, true, false, true, false);
+        ShardingSphereColumn column2 = new ShardingSphereColumn("foo_column_2", Types.INTEGER, false, true, false, true, false);
+        shardingSphereTable.putColumn(column1);
+        shardingSphereTable.putColumn(column2);
+
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:45]: Please remove useless blank line.

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        assertThat(shardingSphereTable.getColumn("foo_column_1"), is(column1));
+        assertThat(shardingSphereTable.getColumn("foo_column_2"), is(column2));
+        assertThat(shardingSphereTable.getColumns(), hasSize(2));
+    }
+    
+    @Test
+    void getColumn() {
+        ShardingSphereColumn column = new ShardingSphereColumn("foo_column", Types.INTEGER, true, true, false, true, false);
+        shardingSphereTable.putColumn(column);
+
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:55]: Please remove useless blank line.

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        assertThat(shardingSphereTable.getColumn("foo_column"), is(column));
+        assertThat(shardingSphereTable.getColumn("FOO_COLUMN"), is(column));
+        assertNull(shardingSphereTable.getColumn("invalid"));
+    }
+    
+    @Test
+    void getColumns() {
+        ShardingSphereColumn column1 = new ShardingSphereColumn("foo_column_1", Types.INTEGER, true, true, false, true, false);
+        ShardingSphereColumn column2 = new ShardingSphereColumn("foo_column_2", Types.INTEGER, false, true, false, true, false);
+
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:65]: Please remove useless blank line.

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    
+    @Test
+    void getColumns() {
+        ShardingSphereColumn column1 = new ShardingSphereColumn("foo_column_1", Types.INTEGER, true, true, false, true, false);
+        ShardingSphereColumn column2 = new ShardingSphereColumn("foo_column_2", Types.INTEGER, false, true, false, true, false);
+        
+        shardingSphereTable.putColumn(column1);
+        shardingSphereTable.putColumn(column2);
+
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:68]: Please remove useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    
+    private ShardingSphereTable shardingSphereTable;
+    
+    @BeforeEach
+    void setUp() {
+        shardingSphereTable = new ShardingSphereTable();
+    }
+    
+    @Test
+    void putColumn() {
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:40]: Please rename test method to assertPutColumn.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Review by strongduanmu — APPROVED** [review]

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        assertThat(shardingSphereTable.getIndexes(), hasSize(2));
+    }
+    
+    @Test
+    void assertContainsIndex() {
+        ShardingSphereIndex index1 = new ShardingSphereIndex("foo_index_1");
+        ShardingSphereIndex index2 = new ShardingSphereIndex("foo_index_2");
+        shardingSphereTable.putIndex(index1);
+        shardingSphereTable.putIndex(index2);
+
```
Comment (strongduanmu) [inline infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java:131]: Please remove useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java (hunk 6 above)
Comment (100paperkite) [reply infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/model/ShardingSphereTableTest.java]: Sorry for missing that. I removed it :)

**Review by strongduanmu — APPROVED** [review]

**Comment by strongduanmu** [comment]
@100paperkite Good job, merged.

**Pull request merged**